app/controllers/candidate_interface/sign_in_controller.rb

```diff
@@ -80,9 +75,38 @@ def authenticate
 
     def expired
       raise unless FeatureFlag.active?('improved_expired_token_flow')
-      return redirect_to candidate_interface_sign_in_path unless params[:id]
 
-      @candidate = Candidate.find(params[:id])
+      if params[:u].blank?
+        redirect_to candidate_interface_sign_in_path
+        return
+      end
+
+      begin
+        candidate_id = Encryptor.decrypt(params[:u])
+        @candidate = Candidate.find(candidate_id)
```

Comment:
I think we don't need to decrypt & find the candidate here, we're not using it for anything?